ARCs/arc-0056.md

```diff
@@ -0,0 +1,376 @@
+---
+arc: 56
+title: Extended App Description
+description: Adds information to the ABI JSON description
+author: Joe Polny (@joe-p), Rob Moore (@robdmoore)
+discussions-to: https://github.com/algorandfoundation/ARCs/issues/258
+status: Draft
+type: Standards Track
+category: ARC
+created: 2023-11-14
+requires: 4, 22, 28
+---
+
+## Abstract
+
+This ARC takes the existing JSON description of a contract as described in [ARC-4](./arc-0004.md) and adds more fields for the purpose of client interaction
+
+## Motivation
+
+The data provided by ARC-4 is missing a lot of critical information that clients should know when interacting with an app. This means ARC-4 is insufficient to generate type-safe clients that provide a superior developer experience.
+
+On the other hand, [ARC-32](./arc-0032.md) provides the vast majority of useful information that can be used to <a href="https://github.com/algorandfoundation/algokit-cli/blob/main/docs/features/generate.md#1-typed-clients">generate typed clients</a>, but requires a separate JSON file on top of the ARC-4 json file, which adds extra complexity and cognitive overhead.
+
+## Specification
+
+### Contract Interface
+
+Every application is described via the following interface which is an extension of the `Contract` interface described in [ARC-4](./arc-0004.md).
+
+```ts
+/** Describes the entire contract. This interface is an extension of the interface described in ARC-4 */
+interface Contract {
+  /** The ARCs used and/or supported by this contract. All contracts implicity support ARC4 and ARC56 */
```

Comment:
```suggestion
  /** The ARCs used and/or supported by this contract. All contracts implicity support ARC-4, ARC-28 and ARC-56 */
```

Reply:
I didn't put ARC-28 and ARC-22 here by default because a contract might not use events or readonly at all, in which case it would seem weird to say every contract supports them. I also don't think it's technically wrong, but if we want to include ARC-28 here we should also include ARC-22.

Reply:
That's fair. Need to keep it in mind for Puya to emit those ARCs if those features are used then. cc @achidlow